starfish/pipeline/filter/gaussian_high_pass.py

```diff
@@ -29,33 +38,33 @@ def get_algorithm_name(cls) -> str:
     @classmethod
     def add_arguments(cls, group_parser: argparse.ArgumentParser) -> None:
         group_parser.add_argument(
-            "--sigma", default=1, type=int, help="standard deviation of gaussian kernel")
+            "--sigma", default=1, type=float, help="standard deviation of gaussian kernel")
 
     @staticmethod
-    def gaussian_high_pass(img: np.ndarray, sigma) -> np.ndarray:
+    def high_pass(image: np.ndarray, sigma: Union[Number, Tuple[Number]]) -> np.ndarray:
         """
         Applies a gaussian high pass filter to an image
 
         Parameters
         ----------
-        img : numpy.ndarray
-            Image to filter
-        sigma : Union[float, int]
+        image : numpy.ndarray
+            2-d or 3-d image data
+        sigma : Union[Number, Tuple[Number]]
             Standard deviation of gaussian kernel
 
         Returns
         -------
-        numpy.ndarray :
-            Filtered image, same shape as input
+        np.ndarray :
+            Filtered image, same shape as input, converted to np.uint16 dtype.
 
         """
-        blurred: np.ndarray = GaussianLowPass.low_pass(img, sigma)
+        blurred: np.ndarray = GaussianLowPass.low_pass(image, sigma)
 
-        over_flow_ind: np.ndarray[bool] = img < blurred
-        res: np.ndarray = img - blurred
-        res[over_flow_ind] = 0
+        over_flow_ind: np.ndarray[bool] = image < blurred
```

Comment:
I don't think this approach works for (-1, +1) ranged data.

Reply:
You're right, it won't, however the gets cast to uint16 in `low_pass`, so that's not a problem here (although the _casting_ is potentially a big problem).

Reply:
... and now that casting is explicit and won't break the results.